server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.database.runs.functions
+
+import doobie.Fragment
+import doobie.implicits.toSqlInterpolator
+import doobie.util.Read
+import play.api.libs.json.Json
+import za.co.absa.atum.model.dto.AdditionalDataSubmitDTO
+import za.co.absa.atum.server.api.database.PostgresDatabaseProvider
+import za.co.absa.atum.server.api.database.runs.Runs
+import za.co.absa.atum.server.model.PartitioningForDB
+import za.co.absa.fadb.DBSchema
+import za.co.absa.fadb.doobie.DoobieFunction.DoobieSingleResultFunctionWithStatus
+import za.co.absa.fadb.doobie.{DoobieEngine, StatusWithData}
+import za.co.absa.fadb.status.handling.implementations.StandardStatusHandling
+import zio._
+import zio.interop.catz._
+
+import doobie.postgres.implicits._
+
+class CreateOrUpdateAdditionalData(implicit schema: DBSchema, dbEngine: DoobieEngine[Task])
+    extends DoobieSingleResultFunctionWithStatus[AdditionalDataSubmitDTO, Unit, Task]
+    with StandardStatusHandling {
+
+  override def sql(values: AdditionalDataSubmitDTO)(implicit read: Read[StatusWithData[Unit]]): Fragment = {
+    val partitioning = PartitioningForDB.fromSeqPartitionDTO(values.partitioning)
+    val partitioningJsonString = Json.toJson(partitioning).toString
```

Comment:
Yeah, I agree with that.

Reply:
It definitely is a bit cleaner, I agree. 

On the other hand, and that is the reason I haven't chosen such approach, is that it would require an additional layer, most likely additional case class containing almost exactly what the DTO contains, and I don't like such duplication. 

If I would do it only for the checkpoint, then we would have only 1 such additional case class. But this is ONLY a 'problem' for partitioning - and partitioning is almost everywhere - additional data, partitioning, checkpoint, and who knows what else in the future. 

I don't want to have almost the exact copy of our DTO model in the database model. Instead, I rather wanted to target only the 'problematic' (=needs converting) one, and that is Partitioning, that's why there is this `PartitioningForDB` functionality.

I was thinking about doing this implicitly or to create some sort of anonymous class that would have this 'for-DB' partitioning already, but I disregarded such solution - again, on one hand it is just 1 line of code.

Reply:
This is basically a 'side effect' (if I can call if that way) of us not doing almost anything on the server (and thus keeping the server code extremely lightweight) - request comes, there is some boilerplate code for ZIO, and then it's slightly adjusted (mostly for SerDe purposes) and then it's send to DB that is doing most heavy lifting. Unless it makes sense, I wouldn't like to keep our approach as is for now